OsmAnd/src/net/osmand/plus/routing/RouteCalculationResult.java

```diff
@@ -395,50 +397,79 @@ public double getRouteVisibleAngle() {
 						// Consider roundabout end.
 						info.routeEndPointOffset = roundAboutEnd;
```

Comment:
This code needs to be moved to backend.
Suggested place - 
RouteSegmentResult

getTurnFollowDestinationRef(String lang, boolean transliterate)

getTurnFollowDestinationNames(String lang, boolean transliterate)

getTurnFollowStreetNames(String lang, boolean transliterate)

getTurnFollowRef(String lang, boolean transliterate)